src/sentry/static/sentry/app/utils/tracing.jsx

```diff
@@ -1,26 +1,46 @@
 import {uniqueId} from './guid';
 
+let tracingEnabled = false;
 let spanId = null;
 let transactionId = null;
+let currentRoute = null;
 
-export function startTransaction() {
-  spanId = uniqueId();
+export function start() {
```

Comment:
This function doesn’t make sense now. Start what? Start Sentry?  If you want to split off Sentry config you don’t need to rewrite the interface of this module. You can just call the span and transaction methods in your event processor. 

As it is now the only change that seems to be relevant in here is the transaction name capture, but I would have that as a parameter of both startTransaction (it can be known by default) as well as a mutation option. You’ll need the same thing with spans in a fully formed API.

Reply:
This is an inference based on the module name, as it's imported as a whole, eg. `tracing.start`. `tracing.startTracing` is redundant, unless it's imported as a standalone function.

Reply:
> You can just call the span and transaction methods in your event processor.

If I'll do this, it won't change on every route change through `this.unlisten = browserHistory.listen(() => tracing.startTransaction());` call (which was the previous behavior and I didn't want to modify it as I didn't know if there's a reason behind this).

Reply:
I guess I can agree on tracing.start, but it doesnt actually do that. It **used** to do that, but now all it does is wire up Sentry's supporting code. The way the API was designed is this:

- on every page load, tracing is started
- on every page change, tracing is started
- when tracing is started, a transaction id/span are generated automatically
- if needed, when a tracing is started, the previous existing transaction/span are closed (not needed by us)

So I guess my concerns are why we're coupling the entire tracing logic entirely to Sentry (or why we actually needed major changes at all here). Right now it seems to be doing the same thing as it did before, but with the addition of the route (which I understand needs a change).

My comment re: eventProcessor, was why couldn't the event processor hook every event and just bind transaction_id/span_id as part of that code path? e.g. it could call tracing.getTransactionID() and tracing.getSpanID() and tracing.getTransactionName().

Reply:
Also I understand why you have to use the event processor now - was confused by your comments before. I still think it could just call the existing functions but I care less about that than I do us tightly coupling Sentry (app) tracing to Sentry (sdk). It may be the only consumer we ever have, but if we wanted another we'd have to effectively restore the unchanged behavior here to ensure transactions are bound uniformly.